### PR TITLE
fix contact display

### PR DESCRIPTION
**Pull request by v1r0x**
check if address is set for current user and only display it if an address exists.

### CODE REVIEW

**Comment by v1r0x** [comment]
should fix #14 and #37

**Comment by Henni** [comment]
Doesn't fix #14 
Instead I now see:
image

**Comment by v1r0x** [comment]
Yes, https://github.com/v1r0x/maps/blob/fix-contacts/js/script.js#L733 doesn't get updated, but all contacts are shown correctly.

**Comment by Henni** [comment]
@v1r0x It only fixes #37 then? I didn't have a closer look at the source of maps yet, so maybe you can describe what this has to do with https://github.com/v1r0x/maps/blob/fix-contacts/js/script.js#L733.

Do you have an idea how to fix #14?

**Comment by v1r0x** [comment]
It basically fixes both. The only thing which is currently broken is the progress bar. I'm not a javascript programmer and don't get the problem (should work as it did before).

The link I provided is the line where the total number of contacts is set. In my PR this is set to 0 (and I don't know why). This is why it displays the "x of 1" and infinty%

File: js/script.js [hunk 1]
```diff
@@ -720,23 +720,25 @@ Array.prototype.unique = function() {
...
+				}
+				var total = Maps.tempTotal;
+				var index = Maps.tempCounter
+				var percent = Math.round((index / total * 100) * 100) / 100;
+				toolKit.setProgress(percent);
+				$('#cCounter').text(index + 1 + ' of ' + (total * 1 + 1));
+				Maps.tempCounter++;
+				if (index == total)
+					$('#loadingContacts').hide()
+				Maps.getContactPositionData();
```
Comment (Henni) [inline js/script.js]: @v1r0x you have to put the lines 733 - 741 into the callback of `toolKit.adresLookup()`

**Comment by v1r0x** [comment]
Is it possible to check if the address is undefined and only do the lookup if it is not?

**Comment by Henni** [comment]
@v1r0x how about checking if the addresse is undefined in the function `toolKit.adresLookup()`?

**Comment by v1r0x** [comment]
Yes, I had a look at the code and this seems good. Thanks for the help!
Should work now

**Comment by Henni** [comment]
@jancborchardt could you please take a look and confirm that this fixes your issue #14 

@v1r0x It would be great if you could fix a small beauty issue. The loading bar should be reset at the beginning of loading the contacts. Otherwise you see the completed loading bar for a short time.

**Comment by jancborchardt** [comment]
WOW! This looks great! And fixes #14 for me. The only issue is that the actual avatars are not shown, just these dark silhouettes for everyone.

But I suppose adding that functionality is for a separate pull request?

**Comment by v1r0x** [comment]
Yes, I think this is another problem (hopefully^^).
Another problem: If a contact has more than one address, only one is shown

**Comment by Henni** [comment]
@jancborchardt yes this should be a different pull request
Thanks for the feedback!

@v1r0x I think it is ok to just show one address per contact, but it is important that the 'preferred' address is chosen.

**Comment by jancborchardt** [comment]
Should we merge it then? :+1: from me. @Henni if you think it’s good press the button. :)

**Comment by Henni** [comment]
Let's fix the loading bar in another PR.

**Pull request merged**